### PR TITLE
feat(jdk15): adapt jdk15 & check nashorn dependency & print maven dep…

### PR DESCRIPTION
**Pull request by Ice2Faith**
- 适配JDK15
- 检查nashorn引擎依赖
- 依赖不存在时，控制台输出需要添加的Maven依赖坐标
- 更改引擎名称从nashorn为JavaScript
- Sm4添加生成秘钥方法
- 重构*.js初始化引擎到Provider类中

### CODE REVIEW

**Comment by Ice2Faith** [comment]
引擎不存在时，打印信息如下：
none nashorn JavaScript engine found! please add this dependency into your pom.xml: 
<dependency>
    <groupId>org.openjdk.nashorn</groupId>
    <artifactId>nashorn-core</artifactId>
    <version>15.4</version>
</dependency>

File: src/main/java/com/antherd/smcrypto/NashornProvider.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package com.antherd.smcrypto;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author Ice2Faith
+ * @date 2024/8/7 19:54
+ * @desc
```
Comment (antherds) [inline src/main/java/com/antherd/smcrypto/NashornProvider.java:8]: Add descriptions or Remove this tag.
If you want to add custom annotation tags, you also need to add maven-javadoc-plugin config options,
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-javadoc-plugin</artifactId>
    <version>2.9.1</version>
    <configuration>
        
        <additionalJOptions>
            <additionalJOption>-Xdoclint:none</additionalJOption>
        </additionalJOptions>
    </configuration>
</plugin>

Comment (geyiwei-hub) [inline src/main/java/com/antherd/smcrypto/NashornProvider.java:8]: Add descriptions or Remove this tag.
If you want to add custom annotation tags, you also need to add maven-javadoc-plugin config options,
```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-javadoc-plugin</artifactId>
    <version>2.9.1</version>
    <configuration>
        <!-- for jdk1.8, remove this on jdk1.7 -->
        <additionalJOptions>
            <additionalJOption>-Xdoclint:none</additionalJOption>
        </additionalJOptions>
    </configuration>
</plugin>
````

File: src/main/java/com/antherd/smcrypto/NashornProvider.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    status.set(true);
+                    break;
+                }
+            } catch (Throwable e) {
+
+            }
+        }
+        checked.set(true);
+        return status.get();
+
```
Comment (geyiwei-hub) [inline src/main/java/com/antherd/smcrypto/NashornProvider.java:69]: Remove this blank line.

File: src/main/java/com/antherd/smcrypto/Provider.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.antherd.smcrypto;
+
+
```
Comment (geyiwei-hub) [inline src/main/java/com/antherd/smcrypto/Provider.java:3]: Remove this blank line.

File: src/main/java/com/antherd/smcrypto/sm4/Sm4.java [hunk 4]
```diff
@@ -1,33 +1,40 @@
...
-            engine.eval(new BufferedReader(new InputStreamReader(inputStream)));
-            invocable = (Invocable) engine;
+            invocable = (Invocable) Provider.getJavaScriptEngine(Provider.SM4_CLASSPATH_RESOURCE_PATH);
         } catch (ScriptException e) {
             e.printStackTrace();
         }
     }
 
+
+    public static String generateKey(){
```
Comment (geyiwei-hub) [inline src/main/java/com/antherd/smcrypto/sm4/Sm4.java:32]: Remove this useless code.

**Review by geyiwei-hub — COMMENTED** [review]
Good job.

**Comment by Ice2Faith** [comment]
Ok, Already Adjusted it.

**Pull request merged**